packages/test/test-version-utils/src/compatConfig.ts

```diff
@@ -170,6 +173,47 @@ if (compatKind !== undefined) {
     configList = configList.filter((value) => compatKind!.includes(value.kind));
 }
 
+function getUnexpectedLogErrorException(logger: ErrorTrackingLogger){
+    const results = logger.results();
+    if(results.unexpectedErrors.length > 0){
+        return new Error(
+            // eslint-disable-next-line max-len
+            `Unexpected Errors in Logs. Use itExpects to specify expected errors:\n${ JSON.stringify(results.unexpectedErrors, undefined, 2)}`);
+    }
+    if(results.expectedNotFound.length > 0){
+        return new Error(
+            `Expected Events not found:\n${ JSON.stringify(results.expectedNotFound, undefined, 2)}`);
+    }
+}
+
+function createExpectsTest(expectedEvents: ITelemetryGenericEvent[], test: Mocha.AsyncFunc){
+    return async function (this:Context){
+        const provider: TestObjectProvider | undefined = this.__fluidTestProvider;
+        assert(provider !== undefined, "Expected __fluidTestProvider on this");
+        try{
+            provider.logger.registerExpectedEvent(... expectedEvents);
+            await test.bind(this)();
+        }catch(error){
+            provider.logger.sendErrorEvent({eventName:"TestException"},error)
+        }
+        const err = getUnexpectedLogErrorException(provider.logger);
+        if(err !== undefined){
+            throw err;
+        }
+    };
+}
+
+export type ExpectsTest = (name: string, expectedEvents: ITelemetryGenericEvent[], test: Mocha.AsyncFunc) => Mocha.Test
+
+export const itExpects: ExpectsTest & {only: ExpectsTest} =
```

Comment:
It looks like the events have to be in the order in which they are generated in the tests. Is that correct?
Could you please add a description of what this function does and what to expect, etc.

Reply:
yeah. i went back and forth on order or not. happy to make it clearer